app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js

```diff
@@ -269,9 +271,26 @@ export function addAdminSidebarSectionLink(sectionName, link) {
   additionalAdminSidebarSectionLinks[sectionName].push(link);
 }
 
-function pluginAdminRouteLinks() {
+function pluginAdminRouteLinks(router) {
   return (PreloadStore.get("visiblePlugins") || [])
-    .filter((plugin) => plugin.admin_route && plugin.enabled)
+    .filter((plugin) => {
+      if (!plugin.admin_route || !plugin.enabled) {
+        return false;
+      }
+
+      // Check if the admin route is valid, if it is not the whole admin
+      // interface can break because of this. This can be the case for things
+      // like ad blockers stopping plugin JS from loading.
+      if (adminRouteValid(router, plugin.admin_route)) {
+        return true;
+      } else {
+        warn(
+          `[AdminSidebar] Could not find admin route for ${plugin.name}, route was ${plugin.admin_route.full_location}`,
```

Comment:
Should we mention ad blocking as a common cause here?

Reply:
Yep good idea, will add